### DISCUSSION TITLE
Not able to get markers working

### DISCUSSION THREAD

**Discussion by MikelElorza** [PicoQuant/snAPI#6] [Q&A] [answered]
I am trying to use snAPI to do Fluorescence Lifetime Imaging (FLIM). For this, I am using the TTL signal from two piezo controllers as markers (one for rows and the other for columns). I have checked the TTL signals in a oscilloscope and they seem to work properly, but when I connect these signals to the HydraHarp 400 I am not able to obtain the row and column signals back. 

I have enabled the markers with 'sn.device.setMarkerEnable(1,1,0,0)' as Im using M1 and M2. When I go to the PTU file and check whether the records are markers or not everything seem to work properly (I get the expected pulses from a 10x10 pixel scan):
![Capture](https://github.com/PicoQuant/snAPI/assets/107248756/8d0306cd-b43f-4fc0-826d-ef93ec9178e3)
However, when I try to obtain the M1 and M2 separately, it seems that it gives me random pulses instead of the regular pulse sequences that I expect from a scan: 
![Capture2](https://github.com/PicoQuant/snAPI/assets/107248756/3ef33bf6-a3b9-4204-b784-abd98580a5b5)

It is also interesting that the amount of pulses that I obtain from M1 and M2 is different than the amount of pulses I get from isMarker().

Thank you for your help

**Comment by tpoint75** [comment, marked as answer]
I am not sure, if the raw data format is the best for analyzing it. The raw data format is compressed, contains overflows and has no absolut times. Please use the Unfold class instead.
Please open a ticket in our support page:
https://www.picoquant.com/contact/support
- add the ptu file
- add the whole python / jupyter script
- and the logfile!

The files can be uploaded there:
https://picoquant.wetransfer.com/

**Reply by MikelElorza** [reply]
Thank you very much! I have tried with 'unfold' instead of 'raw' and I am still having problems. I have just uploaded the notebooks (one for acquisition and another one for analysis, where I try to get the markers back). Lets see if you can help me. Thank you very much, again.

**Reply by MikelElorza** [reply]
I have been playing around with the unfold and it seems that it is working now (I was doing one thing wrong...). Thanks for the attention and sorry the inconvenience. :)

**Reply by tpoint75** [reply]
What have you made wrong? Can we do something to prevent this?

**Reply by MikelElorza** [reply]
It seems that markers function is working fine with the unfold class, but I would say that the raw.markers() function is not working correctly. Maybe you would like look at it or clarify the documentation (for me it wasn´t clear why it shouldn´t work properly with raw class). 
Thank you

**Reply by tpoint75** [reply]
Have you uploaded the script and the ptu file yet?

**Reply by MikelElorza** [reply]
I have just prepared a notebook where you can see the difference between using raw and unfold classes: https://github.com/MikelElorza/Data-analysis/blob/main/Narrow%20field/Imaging/Markers_unfold_raw.ipynb
You can get the data from: https://github.com/MikelElorza/Acquisition2/blob/main/Data/Trial_markers/T/PTU_6.ptu
I hope you can see it correctly.

**Reply by tpoint75** [reply]
You where absolutely right! I found the bug! I have to test it an make an update for snAPI next week! Thank you!
